### DISCUSSION TITLE
hx-boost and relative URLs

### DISCUSSION THREAD

**Discussion by maiksch** [bigskysoftware/htmx#1991] [Q&A] [answered]
I have encountered an interesting problem. First, imagine following URL structure:

- /admin
  - /admin/[id] (this redirects to /admin/[id]/scores)
  - /admin/[id]/scores
  - /admin/[id]/answers

I configured all 3 pages to have `hx-boost="true"` on the body tag and use boosted anchor tags for all my navigation.

Here are the simple steps to reprodure my problem: 

0. I load my page on `/admin`
1. I navigate to `admin/[id]` (notice how I use a relative path)
2. I get navigated to `/admin/[id]/scores` successfully
3. I click on a link in the tab bar to navigate "horizontally" to `answers` (The browser shows `/admin/[id]/answers` when hovering over the anchor tag)
4. I get a 404, because HTMX tries to load `/answers` instead of `/admin/[id]/answers`

Also the other way around:

0. I load my page on `/admin/[id]/answers`
1. I navigate to `/admin` (this time with an absolute path)
2. I get navigated to `/admin` successfully
3. I click on a link to `admin/[id]` (now a relative path again)
4. I get a 404, because HTMX tries to load `/admin/q/[id]/admin/q/[id]` (huh?)

Did I miss something in the documentation regarding hx-boost and relative URLs? Because if I remove the `hx-boost="true"` from my links that navigate "vertically" instead of "horizontally" everything works just fine. Is this type of link structure not possible with boosted links?

**Comment by croxton** [comment, marked as answer]
I think this is because htmx will use the literal value of the anchor href for the request, rather than the fully-resolved url of the anchor provided by the browser (what you see when you hover over the link). I did introduce fully-resolved hrefs for request URLs in [this PR](https://github.com/bigskysoftware/htmx/pull/1338) but that change was [subsequently reverted](https://github.com/bigskysoftware/htmx/pull/1742) for the reasons discussed in that thread.

If you are careful in constructing the urls in your boosted anchors so they are always _root-relative_ or _absolute_, then you shouldn't encounter this problem.
